ml_service/pipelines/diabetes_regression_build_parallel_batchscore_pipeline.py

```diff
@@ -0,0 +1,461 @@
+) -> Datastore:
+    """
+    Obtains a datastore with matching name. Creates it if none exists.
+
+    :param datastorename: Name of the datastore
+    :param ws: Current AML Workspace
+    :param env: Environment variables
+    :param input: Datastore points to the input container if
+    this is True(default) or the output storage container otherwise
+
+    :returns: Datastore
+
+    :raises: ValueError
+    """
+    if datastorename is None:
+        raise ValueError("Datastore name is required.")
+
+    containername = (
+        env.scoring_datastore_input_container
+        if input
+        else env.scoring_datastore_output_container
+    )
+
+    if datastorename in ws.datastores:
+
+        datastore = ws.datastores[datastorename]
+
+    # the datastore is not registered but we have all details to register it
+    elif (
+        env.scoring_datastore_access_key is not None
+        and containername is not None  # NOQA: E501
+    ):  # NOQA:E501
+
+        datastore = Datastore.register_azure_blob_container(
+            workspace=ws,
+            datastore_name=datastorename,
+            account_name=env.scoring_datastore_storage_name,
+            account_key=env.scoring_datastore_access_key,
+            container_name=containername,
+        )
+    else:
+        raise ValueError(
+            "No existing datastore named {} nor was enough information supplied to create one.".format(  # NOQA: E501
+                datastorename
+            )
+        )
+
+    return datastore
+
+
+def get_input_dataset(ws: Workspace, ds: Datastore, env: Env) -> Dataset:
+    """
+    Gets an input dataset wrapped around an input data file. The input
+    data file is assumed to exist in the supplied datastore.
+
+
+    :param ws: AML Workspace
+    :param ds: Datastore containing the data file
+    :param env: Environment variables
+
+    :returns: Input Dataset
+    """
+
+    scoringinputds = Dataset.Tabular.from_delimited_files(
+        path=DataPath(ds, env.scoring_datastore_input_filename)
+    )
+
+    scoringinputds = scoringinputds.register(
+        ws,
+        name=env.scoring_dataset_name,
+        tags={"purpose": "scoring input", "format": "csv"},
+        create_new_version=True,
+    ).as_named_input(env.scoring_dataset_name)
+
+    return scoringinputds
+
+
+def get_fallback_input_dataset(ws: Workspace, env: Env) -> Dataset:
+    """
+    Called when an input datastore does not exist or no input data file exists
+    at that location. Picks up an included test data file from the "data"
+    location in the source code repo named "pcamvpscoringinput.csv" and
+    returns a dataset wrapped around that file. Useful when debugging
+    this code in the absence of the input data location Azure blob.
+
+
+    :param ws: AML Workspace
+    :param env: Environment Variables
+
+    :returns: Fallback input dataset
+
+    :raises: FileNotFoundError
+    """
+
+    file_path = os.path.join("data", env.scoring_datastore_input_filename)
+
```

Comment:
This would be the first time we use the data folder in this way. Currently we get the diabetes data from [here](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_diabetes.html), and [load it to the datastore if needed](https://github.com/microsoft/MLOpsPython/blob/cbeadae5b53409f71ff69a34c27ed4aa1e6606da/ml_service/pipelines/diabetes_regression_build_train_pipeline.py#L62). The data folder is not meant to hold data for training.

I am hesitant on this approach since the user can place their data in the repo, and it will be picked up.

@shivp950 Any thoughts?

Reply:
@j-so that's a fair comment. I will work on changing my approach to match the existing one. In progress.